pallets/committee/src/lib.rs

```diff
@@ -524,7 +537,12 @@ pub mod pallet {
 				Error::<T>::InvalidVotingPeriod
 			);
 
-			VotingPeriod::<T>::set(voting_period);
+			if Proposals::<T>::iter_values().filter(|p| p.status == ProposalStatus::Active).count().is_zero() {
+				VotingPeriod::<T>::set(voting_period);
```

Comment:
I think we should always wait until the current voting period is over and use always the pending storage